app/src/main/java/com/plaid/linksample/MainActivity.kt

```diff
@@ -13,38 +13,22 @@ import android.view.View
 import android.widget.TextView
```

Comment:
This is essentially the previous `MainActivityResultContract`, but with a the success/error case extracted into a separate function to keep things cleaner + with the options menu to switch to other examples.